TypeSystem/flow-analysis/reachability_A20_t03.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion - Binary operator: All binary operators other than `==`, `&&`,
+/// `||`, and `??` are handled as calls to the appropriate `operator` method.
+///
+/// @description Checks that if the static type of the expression of the form
+/// `E1 + E2` is `Never` `after(N) = unreachable(after(E2))`.
+/// @author sgrekhov22@gmail.com
+
+class C<T extends Never> {
+  T operator +(int other) => throw "";
+}
+
+class C2<T extends Never> {
+  Future<T> operator +(int other) async => throw "";
+}
+
+void test1() {
+  late int i;
+  if (2 > 1) {
+    C() + 0;
+    i = 42;
+  }
+  i; // Definitely unassigned
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+void test2() {
+  late int i;
+  if (2 > 1) {
+    C2() + 0;
```

Comment:
Wouldn't this make the program throw after `main` has returned? So we're relying really heavily on the fact that this program won't run (because it has errors).

Perhaps use some other type than `Future<Never>` when the point is simply that the type isn't `Never`.

Reply:
I'm just (slightly) worried because it is a source of confusion: Comment out a line that has a compile-time error and run the test. In this case the test will get the static checking, and it may also be executed (if there are no other compile-time errors left). However, it will now throw, because of a `Future<Never>` which will be completed after `main` returns.